src/UI/Actions.hs

```diff
@@ -585,11 +579,14 @@ openAttachment =
 openWithCommand :: Action 'ViewMail 'MailAttachmentOpenWithEditor (T.Next AppState)
 openWithCommand =
   Action
-  { _aDescription = ["ask for command to open attachment"]
-  , _aAction = \s ->
-    let cmd = view (asMailView . mvOpenCommand . E.editContentsL . to (T.unpack . currentLine)) s
-    in Brick.suspendAndResume $ liftIO $ openCommand' s cmd
-  }
+    { _aDescription = ["ask for command to open attachment"]
+    , _aAction =
+        \s ->
+          let cmd = view (asMailView . mvOpenCommand . E.editContentsL . to (T.unpack . currentLine)) s
+           in if null cmd
+                then Brick.continue $ setError (GenericError "Empty command") s
+                else Brick.suspendAndResume $ liftIO $ openCommand' s (MailcapHandler (Process (fromList cmd) []) False)
```

Comment:
Rather than use `null` and the non-total `fromList`, instead:
```haskell
case cmd of
  [] -> error case
  x:xs -> blah blah (Process (x :| xs)) blah
```

Reply:
True!